PWGHF/D2H/Tasks/taskOmegacSt.cxx

```diff
@@ -0,0 +1,166 @@
+          o2::base::Propagator::initFieldFromGRP(grpmag);
+        } else {
+          LOG(fatal) << "Got nullptr from CCDB for path " << grpmagPath << " of object GRPMagField and " << grpPath << " of object GRPObject for timestamp " << run3grp_timestamp;
+        }
+      }
+    }
+
+    const auto primaryVertex = getPrimaryVertex(collision);
+    o2::dataformats::DCA impactParameterTrk;
+    for (const auto& trackedCascade : trackedCascades) {
+      const auto trackCasc = trackedCascade.track_as<TracksExt>();
+      auto trackParCovTrk = getTrackParCov(trackCasc);
+      // trackParCovTrk.propagateToDCA(primaryVertex, bz, &impactParameterTrk);
+      // gpu::gpustd::array<float, 2> dcaInfo;
+      o2::base::Propagator::Instance()->propagateToDCABxByBz(primaryVertex, trackParCovTrk, 2.f, static_cast<o2::base::Propagator::MatCorrType>(cfgMaterialCorrection.value), &impactParameterTrk);
+
+      const auto& casc = trackedCascade.cascade();
+      const auto& bachelor = casc.bachelor_as<TracksExt>();
+      const auto& v0 = casc.v0();
+      const auto& ptrack = v0.posTrack_as<TracksExt>();
+      const auto& ntrack = v0.negTrack_as<TracksExt>();
+
+      registry.fill(HIST("hDca"), std::sqrt(impactParameterTrk.getR2()));
+      registry.fill(HIST("hDcaXY"), impactParameterTrk.getY());
+      registry.fill(HIST("hDcaZ"), impactParameterTrk.getZ());
+      registry.fill(HIST("hDcaVsPt"), impactParameterTrk.getY(), trackCasc.pt());
+      registry.fill(HIST("hDcaVsR"), impactParameterTrk.getY(), RecoDecay::sqrtSumOfSquares(trackCasc.x(), trackCasc.y()));
+      registry.fill(HIST("hMassVsPt"), trackedCascade.omegaMass(), trackCasc.pt());
+
+      if (!ptrack.has_mcParticle() || !ntrack.has_mcParticle() || !bachelor.has_mcParticle())
+        continue;
+
+      LOGF(debug, "ptrack (id: %d, pdg: %d) has mother %d", ptrack.mcParticleId(),
+           ptrack.mcParticle().pdgCode(), ptrack.mcParticle().has_mothers() ? ptrack.mcParticle().mothersIds()[0] : -1);
+      LOGF(debug, "ntrack (id: %d, pdg: %d) has mother %d", ntrack.mcParticleId(),
+           ntrack.mcParticle().pdgCode(), ntrack.mcParticle().has_mothers() ? ntrack.mcParticle().mothersIds()[0] : -1);
+
+      LOG(debug) << "bachelor with PDG code: " << bachelor.mcParticle().pdgCode();
+      if (ptrack.mcParticle().has_mothers() && ntrack.mcParticle().has_mothers() &&
+          ptrack.mcParticle().mothersIds()[0] == ntrack.mcParticle().mothersIds()[0]) {
+        const auto v0part = ptrack.mcParticle().mothers_as<aod::McParticles>()[0];
+        LOG(debug) << "v0 with PDG code: " << v0part.pdgCode();
+        if (v0part.has_mothers() && bachelor.mcParticle().has_mothers() &&
+            v0part.mothersIds()[0] == bachelor.mcParticle().mothersIds()[0]) {
+          const auto mother = v0part.mothers_as<aod::McParticles>()[0];
+          const auto pdgCode = mother.pdgCode();
+          LOG(debug) << "cascade with PDG code: " << pdgCode;
+          if (std::abs(pdgCode) == 3334) {
```

Comment:
Do not use magic numbers. Use a named PDG value. Define it in `https://github.com/AliceO2Group/O2Physics/blob/master/PWGHF/Core/SelectorCuts.h#L29` if needed.